pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -91,6 +92,15 @@ type Plugin interface {
 	Name() string
 }
 
+// QueueSortPlugin is an interface that must be implemented by "QueueSort" plugins.
+// These plugins are used to sort pods in the scheduling queue. Only one queue sort
+// plugin may be enabled at a time.
+type QueueSortPlugin interface {
+	Plugin
+	// Less are used to sort pods in the scheduling queue.
+	Less(*internalqueue.PodInfo, *internalqueue.PodInfo) bool
```

Comment:
Plugins will not be able to use internalqueue.PodInfo if they import scheduler's code. We should probably move PodInfo outside of the internal.

cc/ @misterikkit

Reply:
I could move the `PodInfo` into this file if it is appropriate.

Reply:
It feels a bit odd to move PodInfo here, but since we want plugins to use it, I guess we don't have any other option.

Reply:
out of the scope of this pr, but @bsalamat does the same logic apply for `NodeInfoSnapshot` used here, which is also under internal?

Reply:
> It feels a bit odd to move PodInfo here, but since we want plugins to use it, I guess we don't have any other option.

Moved LessFunc and PodInfo to the framework interface.